### ISSUE TITLE
Clusters in HitMapView

### ISSUE THREAD

**Issue by anjomshoaa** [sevamoo/SOMPY#24]
Hi, 

I have a question about the HitMapView. I do clustering as follows and the output is perfectly fine: 

   cl = som.cluster(n_clusters=3)

But the HitMapView always uses the default number of clusters which is 8 and I could not find a way to change it: 

h = hitmap.HitMapView(10, 10, 'hitmap', text_size=8, show_text=True)

How can I specify number of clusters in a HitMapView to a different number? 

Regards
Amin

**Comment by sevamoo** [comment]
Hi @anjomshoaa there was an issue in som.cluster() that it didn't set the cluster labels. Now should be fine. 
Could you please check it out?

**Comment by anjomshoaa** [comment]
Thanks Vahid jan, 

Yes, it is working perfectly now. 

Best regards
Amin

**Comment by gsvijayraajaa** [comment]
Is there a way to get the index of the training data for the individual cluster labels?

**Comment by sevamoo** [comment]
by clustering you get the labels of nodes. By project function you get the bmus. So, then use something like this
cl = som.cluster(n_clusters=3)
bmus = som.project()
cl_data = cl[bmus]

**Comment by gsvijayraajaa** [comment]
Thanks for the direction.

Is there a som.project() function available?

Regards,
Vijay Raajaa GS

**Comment by sevamoo** [comment]
Yes, there is one. 
https://github.com/sevamoo/SOMPY/blob/c673b194079a846f974994bda2b3238d3c757736/sompy/sompy.py#L433

**Comment by gsvijayraajaa** [comment]
Using the project_data function , I need to project back the training data to the trained model?

**Comment by vishvanath45** [comment]
@gsvijayraajaa Did you get a way get the output category label back to data.

**Comment by vedavikas06** [comment]
# to cluster the neurons into 4 clusters
cl = som.cluster(n_clusters=4)
#this will map the data X to the respective neurons in som 
bmus = som.project_data(X)
print(bmus)
# this gives the corresponding cluster number for the data in X
cl_data = cl[bmus]
print(cl_data)

**Comment by vishvanath45** [comment]
Thanks it worked!

**Comment by businessglitch** [comment]
Hey @sevamoo, really need help with this issue. I have a dataset of length 50,000 when i do cl = som.cluster(n_clusters=2)

cl returns lables which is fine but the length of 1120.  Which is understandable because the data points are under a cluster.

but when I do  cl_data = cl[bmus], this creates duplicate indexs like there would be 7 1's  5 2's and so on, which is not correct because the model wouldn't use duplicated values. Any advice on this would be appreciated. I believe it has something to do with the length of cl the 1120

**Comment by sevamoo** [comment]
I believe this is the answer you are looking for: cl_data = cl[bmus]

**Comment by kedarnaik33** [comment]
Hi Sevamoo, 

I have a question, in the som.cluster(n_clusters = 8), I have set 8 clusters and it gives me clusters from 0 to 7. 

However, my test set labels are from 1 to 8 and when I use accuracy_score it obviously gives me a wrong accuracy. 

I wanted to know if there is a way to set the cluster number from 1 to 8 ?

Would really appreciate your help. 

Sincerely, 
Kedar

**Comment by akol67** [comment]
> 
> 
> Using the project_data function , I need to project back the training data to the trained model? Also, from the bmus, how do I tie it back to the original training data index? Essentially I would like to find the list of training data index corresponding to the cluster label.

Did you have some help to accomplish that list of training index?
